virttest/storage.py

```diff
@@ -393,19 +391,18 @@ def get_backup_set(filename, backup_dir, action, good):
                 elif action == 'restore':
                     bkp_set = [[dst, src]]
             else:
-                # We have to make 2 backups, one of the bad image, another one
-                # of the good image
+                # When error occured,
+                # for backup: it keeps the image for further debug,
+                # for restore: it replaces the bad image with the good one.
                 src_bad = filename
-                src_good = os.path.join(backup_dir, "%s.backup" % basename)
-                hsh = utils_misc.generate_random_string(4)
+                hsh = utils_misc.generate_random_string(6)
+                good_bkp = os.path.join(backup_dir, "%s.backup" % basename)
                 dst_bad = (os.path.join(backup_dir, "%s.bad.%s" %
                                         (basename, hsh)))
-                dst_good = (os.path.join(backup_dir, "%s.good.%s" %
-                                         (basename, hsh)))
                 if action == 'backup':
-                    bkp_set = [[src_bad, dst_bad], [src_good, dst_good]]
+                    bkp_set = [[src_bad, dst_bad]]
```

Comment:
well I like you're not reverting the image as that should be done automatically via `restore_image_after_testing`, the only problem is that some people might have this disabled and then this is a change of behaviour. To me this is not a big issue, just mentioning the possible consequences.

Reply:
(I mean people can use `restore_image_after_testing=no restore_image_after_testing_on_error=yes` that would result into the old behaviour. Maybe would you please add a note to the commit saying this?